akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -753,6 +768,158 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
+        val request = HttpRequest(uri = "https://www.example.com/")
+        user.emitRequest(request)
+        user.emitRequest(request)
+        user.emitRequest(request)
+        // ... and backpressured requests...
+        user.emitRequest(request) // this emit succeeds but is buffered
+
+        network.expect[HeadersFrame]().streamId shouldBe (1)
+        network.expect[HeadersFrame]().streamId shouldBe (3)
+        network.expect[HeadersFrame]().streamId shouldBe (5)
+        network.expectNoBytes(100.millis)
+
+        // ... and some request already got their response (canary round-trip)
+        network.sendFrame(HeadersFrame(streamId = 1, endStream = true, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None))
+        val resp1: HttpResponse = user.expectResponse()
+
+        // ... and a completed user handler.
+        user.requestOut.sendComplete()
+
+        // When the first few response-frames arrive...
+        network.sendFrame(HeadersFrame(streamId = 3, endStream = true, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None))
+        // Then the backpressured in-flight streams proceed (sending pending requests)...
+        network.expect[HeadersFrame]().streamId shouldBe (7)
+        network.sendFrame(HeadersFrame(streamId = 5, endStream = false, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None))
+        network.sendFrame(DataFrame(streamId = 5, endStream = true, ByteString("Hello World 5!")))
+        network.sendFrame(HeadersFrame(streamId = 7, endStream = false, endHeaders = true, HPackSpecExamples.C61FirstResponseWithHuffman, None))
+        network.sendFrame(DataFrame(streamId = 7, endStream = true, ByteString("Hello World 7!")))
+        //... and it is possible to consume all responses (from stream 1 to stream 7)
+        val resp3: HttpResponse = user.expectResponse()
+        val resp5: HttpResponse = user.expectResponse()
+        val resp7: HttpResponse = user.expectResponse()
+
+        private val eventualEntities: Seq[Future[HttpEntity.Strict]] =
+          Seq(resp1, resp3, resp5, resp7).map(_.entity.toStrict(100.millis))
+        private val entities: Seq[HttpEntity.Strict] = Future.sequence(eventualEntities).futureValue
+        // The complete entity of the responses is available
+        entities shouldBe Seq(
+          HttpEntity.Empty, // 1 has no data
+          HttpEntity.Empty, // 3 has no data
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString("Hello World 5!")),
+          HttpEntity.Strict(ContentTypes.`application/octet-stream`, ByteString("Hello World 7!"))
+        )
+
+        // Completion doesn't happen until entities are consumed
+        user.responseIn.expectComplete()
+
+      }
+
+      "until in-flight streaming requests are fully sent (after response is already complete) (uncommon)" inAssertAllStagesStopped new TestSetup(
+        Setting(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 3)
```

Comment:
we can avoid this SETTING for this test.

Reply:
:+1: